Dockerfile

```diff
@@ -1,35 +1,87 @@
-FROM r-base:4.0.3
-
-MAINTAINER https://github.com/pacificclimate/chickadee
-LABEL Description="chickadee WPS" Vendor="Birdhouse" Version="0.1.0"
+FROM rocker/r-ver:4.0.3 AS builder
 
 ENV PIP_INDEX_URL="https://pypi.pacificclimate.org/simple/"
 
-WORKDIR /code
-
 COPY requirements.txt r_requirements.txt install_pkgs.R ./
 
 # Install python and R packages
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends \
+    build-essential \
+    gcc \
+    libpq-dev \
+    python3.8 \
+    python3-pip \
+    python3-setuptools \
+    python3-dev \
+    libssl-dev \
+    libxml2-dev \
+    libudunits2-dev \
+    libnetcdf-dev \
+    libgit2-dev && \
+    Rscript install_pkgs.R r_requirements.txt && \
+    pip3 install --upgrade pip && \
+    pip3 install -r requirements.txt --ignore-installed --user && \
+    pip3 install gunicorn --user
+
+FROM rocker/r-ver:4.0.3 AS prod
+MAINTAINER https://github.com/pacificclimate/chickadee
+LABEL Description="chickadee WPS" Vendor="Birdhouse" Version="0.1.0"
+
 RUN apt-get update && \
     apt-get install -y --no-install-recommends \
-      build-essential \
-      gcc \
-      libgit2-dev \
-      libpq-dev \
       python3.8 \
-      python3-pip \
-      python3-setuptools \
-      python3-dev \
-      libssl-dev \
-      libxml2-dev \
-      libudunits2-dev \
-      libnetcdf-dev && \
-    pip3 install --upgrade pip && \
-    pip3 install -r requirements.txt --ignore-installed && \
-    pip3 install gunicorn && \
-    Rscript install_pkgs.R r_requirements.txt
+      python3-pip
+
+COPY --from=builder /root/.local /root/.local
+
+# Copy compiled library files
+COPY --from=builder /usr/lib/x86_64-linux-gnu/libnetcdf.so.15 \
+  /usr/lib/x86_64-linux-gnu/libhdf5_serial_hl.so.100 \
+  /usr/lib/x86_64-linux-gnu/libhdf5_serial.so.103 \
+  /usr/lib/x86_64-linux-gnu/libcurl-gnutls.so.4 \
+  /usr/lib/x86_64-linux-gnu/libsz.so.2 \
+  /usr/lib/x86_64-linux-gnu/libaec.so.0 \
+  /usr/lib/x86_64-linux-gnu/libudunits2.so.0 \
+  /usr/lib/x86_64-linux-gnu/
```

Comment:
at the bottom here you have `/usr/lib/x86_64-linux-gnu/`, would that not just encompass all of the paths above it? Also how did you come up with this list for future reference (it would be helpful to add the explanation to the comment)?

Reply:
The last line `/usr/lib/x86_64-linux-gnu/` is the destination of the copy statement. So all of the files listed above will be copied to the directory. To obtain the list, I had to run every process and observed what files are needed to run each of them.

Reply:
Oof that is a pretty ugly process. What if you just copied all the content of `/usr/lib/x86_64-linux-gnu/`? or is there a ton of stuff in there?

Reply:
Hmm, I will see how big the difference is, and if there isn't much of a difference in size, I can just copy the whole directory.

Reply:
```
REPOSITORY              TAG       IMAGE ID       CREATED          SIZE
chickadee_multi_stage   latest    354df299751b   21 seconds ago   1.5GB
```
Copying the whole directory seems quite heavy, so I am gonna stick to what I have now.